### PR TITLE
Fix/open office gsub replacement

### LINKED ISSUE #509: can't modify frozen String (FrozenError) in open_office.rb
### Steps to reproduce
require 'roo'
require 'rspreadsheet'

fname = 'test_apos.ods'
wb = Rspreadsheet.new
sheet = wb.create_worksheet('Sheet1')
sheet[1,1] = ''   ### Note : The cell contain empty string
wb.save(fname)

wb = Roo::Spreadsheet.open(fname)
pp wb.last_row
wb.close()

### Issue
I write ods file using rspreadsheet gem. When I read back the file using roo, If any cell value contain empty string , then I get following error
..gems/roo-2.8.2/lib/roo/open_office.rb|520| in `gsub!': can't modify frozen String (FrozenError)

### System configuration
**Roo version**: 2.8.2

**Ruby version**:2.6.1

### PR DESCRIPTION
**Pull request by patrickkulling**
### Summary

Resolves #509 

@kakubin could you prioritize this fix please?

### CODE REVIEW

File: test/test_roo.rb [hunk 1]
```diff
@@ -282,6 +282,12 @@ def test_cell_multiline
     end
   end
 
+  def test_apostrophe_replacement
+    with_each_spreadsheet(:name=>'apostrophe', :format=>[:openoffice]) do |oo|
```
Comment (kakubin) [inline test/test_roo.rb:286]: Please add apostrophe to `test/files/apostrophe.ods`

Comment (patrickkulling) [reply test/test_roo.rb:286]: Hey @kakubin
please have another look at the commit.
The file is exactly here: `test/files/apostrophe.ods`

Are you having trouble running the tests?

Comment (kakubin) [reply test/test_roo.rb:286]: I opened the file in local and it didn't contain an apostrophe.

Comment (patrickkulling) [reply test/test_roo.rb:286]: Yes this is correct. The issue itself is not really due to having an apostrophe in the file, but the `gsub!` method that is forcing to replace one just in case.

It can be reproduced with an .ods file that has an empty string `''` which is the most minimal step to reproduce the issue with the frozen string literal.

Comment (patrickkulling) [reply test/test_roo.rb:286]: @kakubin I am also open to rename this test to something along the lines of "html escape" or "frozen string" or "gsub replacement" if this might be better suitable for this edge case.

WDYT ?

Comment (patrickkulling) [reply test/test_roo.rb:286]: I would recommend renaming this test case and the file to `frozen_string_gsub_replacement` to make it more clear that this is testing a very specific case. 

There should be a separate test for the `&apos; > '` replacement, I agree. Problem is once I save it with my version of open office than for some reason I won't be able to reproduce the actual frozen string issue anymore.

Therefor I would:
1. Rename this test case to make it's existence more clear
2. Create a separate test case to just these the apostrophe replacement

WDYT ?

**Review by kakubin — APPROVED** [review]
sorry for being late :(
It looks good to me to be declarative for what each of these are testing.
